bigbluebutton-html5/imports/ui/components/video-preview/virtual-background/component.jsx

```diff
@@ -189,20 +240,108 @@ const VirtualBgSelector = ({
+              const label = `Background ${imageIndex}`;
+
+              return (
+                <Styled.ThumbnailButtonWrapper key={`${filename}-${index}`}>
+                  <Styled.ThumbnailButton
+                    style={{ backgroundImage: `url(${data})` }}
+                    id={`${filename}-${imageIndex}`}
+                    label={label}
+                    tabIndex={disabled ? -1 : 0}
+                    role="button"
+                    aria-label={label}
+                    aria-describedby={`vr-cam-btn-${imageIndex}`}
+                    aria-pressed={currentVirtualBg?.name?.includes(filename)}
+                    hideLabel
+                    ref={ref => inputElementsRef.current[index + IMAGE_NAMES.length + 1] = ref}
+                    onClick={() => {
+                      const node = findDOMNode(inputElementsRef.current[index + IMAGE_NAMES.length + 1]);
+                      node.focus();
+                      node.click();
+                      _virtualBgSelected(
+                        EFFECT_TYPES.IMAGE_TYPE,
+                        filename,
+                        imageIndex - 1,
+                        { file: data },
+                      );
+                    }}
+                    disabled={disabled}
+                    isVisualEffects={isVisualEffects}
+                  />
+                  <Styled.ButtonWrapper>
+                    <Styled.ButtonRemove
+                      label={intl.formatMessage(intlMessages.removeLabel)}
+                      aria-label={intl.formatMessage(intlMessages.removeLabel)}
+                      data-test="removeCustomBackground"
+                      icon="close"
+                      size="sm"
+                      color="dark"
+                      circle
+                      hideLabel
+                      onClick={() => {
+                        dispatch({
+                          type: 'delete',
+                          uniqueId,
+                        });
+                      }}
+                    />
+                  </Styled.ButtonWrapper>
+                  <div aria-hidden className="sr-only" id={`vr-cam-btn-${imageIndex}`}>
+                    {label}
+                  </div>
+                </Styled.ThumbnailButtonWrapper>
+              );
+            })}
+
+          {isVisualEffects && (
+            <>
+              <Styled.BgCustomButton
+                icon='plus'
+                label={intl.formatMessage(intlMessages.customLabel)}
+                aria-describedby={`vr-cam-btn-custom`}
+                hideLabel
+                tabIndex={disabled ? -1 : 0}
+                disabled={disabled}
+                onClick={() => {
+                  if (customBgSelectorRef.current) {
+                    customBgSelectorRef.current.click();
+                  }
+                }}
+                isVisualEffects={isVisualEffects}
+              />
+              <input
```

Comment:
There's no sanitizing here as to file type, extension, size etc
That allows any file type to be uploaded by the end user - and that may cause undefined behavior in the segmentation worker and in the video-preview selector itself.
The video-list-item variant aborts if the file type is not an image - which is a start. Something similar should be done here - preferably adding the `accept` attribute for filtering purposes in the file and other necessary precautions.

Reply:
It's solved for this PR, but please remember to add an UI feedback (eg a toast) in a follow up when the user tries to upload an invalid file (eg an 8 mb image) - and maybe also log it. Thanks.